### ISSUE TITLE
validateAfterLeaveOrPressedEnter?

### ISSUE THREAD

**Issue by ericpyle** [icebob/vue-form-generator#309]
I'm a little confused by the current field validation options. 

There is validateAfterLoad, and validateAfterChanged (as user types).

How can I validate a field only after a user leaves the control or presses enter (which I believe is the typical way that field validators work?)  I don't really want to bother the user that what they are typing is invalid until they are finished typing it and ready to move on.

Thanks for any help!

**Comment by ericpyle** [comment]
Also, if I am asynchronisly validating a text box as the user types, how do I cancel all but the last validation when the user stops typing?

**Comment by ericpyle** [comment]
Perhaps there is a way for vfg to integrate https://lodash.com/docs#debounce for text inputs ?

**Comment by icebob** [comment]
You should integrate debounce to your async validator.

**Comment by ericpyle** [comment]
@icebob thanks for that suggestion. The examples I had seen with debouncer were trying to control data, but using it to just control the validation should work.

**Comment by ericpyle** [comment]
@icebob having some success, but not quite with _debounce(). Now the error message does not get reported until after I start typing a second time.  Is the _debounce() promise not quite what you expected or am I doing something not right?

```
validator: _.debounce(async function(value, field, model){
                            try {
                                var response = await remote.get(`/validate/${value}`);
                                console.log(response);
                            } catch (error) {
                                console.log(error);
                                return [error.responseJSON.error_message];
                            } finally{
                                this.$root.validating = false;
                            }
                            return [];
                        }, 500)
```

After typing 123456, I hit a breakpoint in my debounce callback, and it returns [error_message]
![image](https://user-images.githubusercontent.com/1125565/31398679-5d38de5e-adaf-11e7-9ab0-c4a9ff96381c.png)

However, I don't see the error message until I start typing again (e.g. "22")

![image](https://user-images.githubusercontent.com/1125565/31398298-175a828a-adae-11e7-9dd4-73f77c4bbe5f.png)

**Comment by icebob** [comment]
Would be good a jsFiddle with this issue in order to we can investigate it.

**Comment by ericpyle** [comment]
@icebob here's a jsFiddle to reproduce the issue
https://jsfiddle.net/ericpyle/rnsgn2ng/

**Comment by icebob** [comment]
Great, thanks!

**Comment by ericpyle** [comment]
Thanks, @icebob 

Also wanted to report that sometimes I still get multiple errors:
![image](https://user-images.githubusercontent.com/1125565/31407327-2cebed52-adca-11e7-8360-34766512a311.png)

**Comment by Alex-Sokolov** [comment]
That's really needed. Now after typing every char starting validation and UI freeze for some time

**Comment by icebob** [comment]
Hi @ericpyle,
I tested and it seems vfg validation can't work properly with lodash debounce. :(

**Comment by icebob** [comment]
I started to add debouncing to vfg in validation process. I think it will solve your problem.
[Fiddle](https://jsfiddle.net/icebob/kdhpdbzr/1/)

**Comment by ericpyle** [comment]
@icebob yes that seems to work just fine and I no longer see multiple errors being reported. how do you recommend I get a version of vfg with the fix?

**Comment by icebob** [comment]
This week I will make a new release.

**Comment by ericpyle** [comment]
@icebob thanks

**Comment by icebob** [comment]
Released in [v2.1.1](https://github.com/icebob/vue-form-generator/releases/tag/v2.1.1)
